### PR TITLE
Merlin

### PR DESCRIPTION
**Pull request by cvravi9**
FOR CONTRIBUTOR:
* [.] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [.] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Review by mtekman — APPROVED** [review]
This is great work -- you've removed a lot of output clutter and tested a wide range of parameters

**Comment by mtekman** [comment]
Some more things that are needed:

http://csg.sph.umich.edu/abecasis/Merlin/reference.html

* -r random seed, this is to ensure reproducibility
* -f allele frequency. This can be estimated via "Maximum Likelihood (-fm)" or "Founders" (-ff) or "All Individuals" (-fa) or "Assume Equal" (-fe) or "User File" (-fe).
   * This should be a \<select\>
   * If the select value is "User File", we need to give users a place to select one of their files.
        * This means the \<select\> needs to be wrapped in a \<conditional\> and the general structure of it is:

```xml
<conditional name="allelefreqs" >
      <param name="mode" type="select"  .... >
             <option value="userfile">User File</option>
              .... other options ...
      </param>
       <when value="something"><!-- do nothing --></when>
       <when value="userfile">
             <!--- set new xml elements here -->
        </when>
</conditional>
```

more examples can be found [here](https://docs.galaxyproject.org/en/master/dev/schema.html#tool-inputs-conditional)

**Comment by mtekman** [comment]
This is great! I think all we need now is to clean up the documentation and we would be ready to merge this with galaxy main :D

So, if you could expand the `label=` with short descriptions of each parameter, and put the long descriptions in the `help=`.

The `<help></help>` needs to be cleaned up so that non-technical users know what the tool does without seeing all the flags.

Finally, the `<citations><citations>` section needs to be populated (see the existing help text for a list of publications.

**Comment by cvravi9** [comment]
> we need a specific package version -- you can find this out by activating the merlin environment in conda, and then doing `conda list` to find out what packages are installed and what versions.

# packages in environment at /home/cvravi9/miniconda3:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    conda-forge
asn1crypto                0.24.0                   py37_0
ca-certificates           2020.1.1                      0
certifi                   2019.11.28               py37_0
cffi                      1.12.2           py37h2e261b9_1
chardet                   3.0.4                    py37_1
conda                     4.8.3            py37hc8dfbb8_0    conda-forge
conda-package-handling    1.6.0            py37h516909a_1    conda-forge
cryptography              2.6.1            py37h1ba5d50_0
idna                      2.8                      py37_0
libedit                   3.1.20181209         hc058e9b_0
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 8.2.0                hdf63c60_1
libstdcxx-ng              8.2.0                hdf63c60_1
ncurses                   6.1                  he6710b0_1
openssl                   1.1.1d               h7b6447c_4
pip                       19.0.3                   py37_0
pycosat                   0.6.3            py37h14c3975_0
pycparser                 2.19                     py37_0
pyopenssl                 19.0.0                   py37_0
pysocks                   1.6.8                    py37_0
python                    3.7.3                h0371630_0
python_abi                3.7                     1_cp37m    conda-forge
readline                  7.0                  h7b6447c_5
requests                  2.21.0                   py37_0
ruamel_yaml               0.15.46          py37h14c3975_0
seqtk                     1.2                           1    bioconda
setuptools                41.0.0                   py37_0
six                       1.12.0                   py37_0
sqlite                    3.27.2               h7b6447c_0
tk                        8.6.8                hbc83047_0
tqdm                      4.43.0                     py_0    conda-forge
urllib3                   1.24.1                   py37_0
wheel                     0.33.1                   py37_0
xz                        5.2.4                h14c3975_4
yaml                      0.1.7                had09818_2
zlib                      1.2.11               h7b6447c_3

**Comment by cvravi9** [comment]
> the `version` here should be the conda package version followed by a `.1`

conda                     4.8.3            py37hc8dfbb8_0    conda-forge
conda-package-handling    1.6.0            py37h516909a_1    conda-forge

File: tools/merlin/merlin.xml [hunk 1]
```diff
@@ -1,8 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
-<tool id="merlin" name="Tool for Linkage Analysis" version="0.1.0" python_template_version="3.5">
-    <!-- Stating the package needed -->
+<!-- Stating the package used along with the version of the Conda -->
+<tool id="merlin" name="Tool for Linkage Analysis" version="4.8.3.1" python_template_version="3.5">
```
Comment (mtekman) [inline tools/merlin/merlin.xml]: "4.8.3.1" → "1.1.2.1"

File: tools/merlin/merlin.xml~ [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Limits : --bits [24], --megabytes, --minutes
+Performance : --trim, --noCoupleBits, --swap, --smallSwap
+Output : --quiet, --markerNames, --frequencies, --perFamily, --pdf,
+--tabulate, --prefix [merlin]
+Simulation : --simulate, --reruns, --save, --trait []
+
+
+FATAL ERROR - The datafile merlin.dat cannot be opened
+ ]]>
+
```
Comment (mtekman) [inline tools/merlin/merlin.xml~:68]: The help text needs to explain what the tool does to users without bombarding them with technical information.

Could you put together a small paragraph on what Merlin is, based on the intro text [here](http://csg.sph.umich.edu/abecasis/merlin/index.html) and the tool abstract [here](https://www.ncbi.nlm.nih.gov/pubmed/11731797)

File: tools/merlin/merlin.xml [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+Common causes for this problem are:
+
+1. You might not have used the correct options to specify input file names, please check the program documentation for information on how to do this
+
+2. The file doesn't exist or the filename might have been misspelt.
+
+3. The file exists but it is being used by another program which you will need to close before continuing
+
+4. The file is larger than 2 GB and you haven't compiled this application with large file support.
```
Comment (mtekman) [inline tools/merlin/merlin.xml]: The help text needs to explain what the tool does to users without bombarding them with technical information.

Could you put together a small paragraph on what Merlin is, based on the intro text [here](http://csg.sph.umich.edu/abecasis/merlin/index.html) and the tool abstract [here](https://www.ncbi.nlm.nih.gov/pubmed/11731797)

**Pull request merged**